src/ipa_pc/mod.rs

```diff
@@ -600,29 +618,62 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
                 Randomness::empty()
             };
 
-            let comm = Self::cm_commit(
-                &ck.comm_key[..(polynomial.degree() + 1)],
-                &polynomial.coeffs,
-                Some(ck.s),
-                Some(randomness.rand),
-            )
-            .into_affine();
+            let key_len = ck.comm_key.len();
+            let p_len = polynomial.coeffs.len();
 
-            let shifted_comm = degree_bound.map(|d| {
-                Self::cm_commit(
-                    &ck.comm_key[(ck.supported_degree() - d)..],
-                    &polynomial.coeffs,
-                    Some(ck.s),
-                    randomness.shifted_rand,
-                )
-                .into_affine()
+            let comm: Vec<G>;
+
+            if p_len > key_len {
+
+                // split poly in segments and commit all of them without shifting
+                comm = (0..p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 }).into_par_iter().map(
```

Comment:
For now - yes, parallelization at this stage is useless. But in case of GPU usage in future single MSM call to GPU doesn't seize the whole device, so we can run multiple MSMs on single GPU device in parallel